datastore/longhorn.go

```diff
@@ -38,7 +40,19 @@ var (
 	VerificationRetryCounts   = 20
 )
 
-func (s *DataStore) InitSettings() error {
+func (s *DataStore) InitSettings(settingPath string) error {
+	defaultSettingMap := map[string]string{}
+	if settingPath != "" {
+		data, err := ioutil.ReadFile(settingPath)
+		if err != nil {
+			logrus.Errorf("Failed to read default setting file %v, will use built-in internal default settings: %v", settingPath, err)
+		} else {
+			if err := yaml.Unmarshal(data, &defaultSettingMap); err != nil {
```

Comment:
From the doc, it seems yaml.Unmarshal can result in a partial result. We shouldn't allow it.

Reply:
If the code fails here, should we continue to initialize the settings?

Reply:
If we cannot find the file, we should fail since it's a user error.